libraries/contracts/LibOrder.sol

```diff
@@ -46,7 +47,7 @@ library LibOrder {
 
     function hashKey(Order memory order) internal pure returns (bytes32) {
         //order.data is in hash for V2 orders
-        if (order.dataType == LibOrderDataV2.V2){
+        if (order.dataType == LibOrderDataV2.V2 || order.dataType == LibOrderDataV3.V3_SELL || order.dataType == LibOrderDataV3.V3_BUY){
```

Comment:
probably we can revert if here? so we hash with data always except older orders
not sure if that's ok. need to think

Reply:
reversed the order of if statements